The Straits Times.js

```diff
@@ -0,0 +1,286 @@
+{
+	"translatorID": "2de01dfe-9572-4775-bf7e-6b55c95d60b0",
+	"label": "The Straits Times",
+	"creator": "Robert Sim",
+	"target": "^https?:\\/\\/(www.)?straitstimes.com\\/",
+	"minVersion": "3.0",
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "gcsibv",
+	"lastUpdated": "2020-12-29 09:10:26"
+}
+
+/*
+	***** BEGIN LICENSE BLOCK *****
+
+	Copyright © 2020 Robert Sim
+
+	This file is part of Zotero.
+
+	Zotero is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
+function detectWeb(doc, url) {
+	if (url.match(/(advertise-with-the-straits-times|newsletter-signup)/)) {
+		return 'webpage';
+	}
+	var pageClass = ZU.xpathText(doc, '//meta[@name="cXenseParse:pageclass"]/@content');
+	if (pageClass === 'article') {
+		return 'newspaperArticle';
+	}
+	if (pageClass === 'frontpage') {
+		return 'multiple';
+	}
+	
+	return false;
+}
+
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) === "multiple") {
+		getMultipleItems(doc, url);
+	} else {
+		scrape(doc, url, detectWeb(doc, url));
+	}
+}
+
+function scrape(doc, url, type) {
+	var newItem = new Zotero.Item(type || 'newspaperArticle');
+	newItem.ISSN = '0585-3923';
+	newItem.url = url;
+	newItem.publicationTitle = 'The Straits Times';
+	newItem.title = ZU.xpathText(doc, '//meta[@name="dcterms.title"]/@content');
+	newItem.abstractNote = ZU.xpathText(doc, '//meta[@name="dcterms.description"]/@content') ? ZU.xpathText(doc, '//meta[@name="dcterms.description"]/@content').replace('Read more at straitstimes.com.', '').trim() : '';
```

Comment:
This could be simplified to

```js
newItem.abstractNote = (ZU.xpathText(doc, '//meta[@name="dcterms.description"]/@content') || '')
	.replace('Read more at straitstimes.com.', '').trim();
```

Reply:
updated